packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -412,7 +412,25 @@ export class ChainService implements ChainServiceInterface {
     contract: Contract,
     channelId: string
   ): Promise<HoldingUpdatedArg> {
+    const blockNumber = await this.provider.getBlockNumber();
     const holding = BN.from(await contract.holdings(channelId));
+    const depositEvents = await contract.queryFilter(
+      {
+        address: contract.address,
+      },
+      blockNumber - this.blockConfirmations - 1
```

Comment:
```suggestion
      blockNumber - this.blockConfirmations + 1
```

Number of confirmations is the number of blocks that have been mined (including the initial block) since this transaction was mined. If the current block is 10 and we require 3 confirmation blocks, we should inspect blocks 8, 9, and 10.